gapis/api/vulkan/profile_static_analysis.go

```diff
@@ -67,6 +103,266 @@ type profilingDataBuilder struct {
 	seenCounters map[counterType]struct{}
 }
 
+func (db profilingDataBuilder) newSampler(idx api.SubCmdIdx) *CounterData {
+	db.data.CounterData = append(db.data.CounterData, CounterData{Index: idx})
+	return &db.data.CounterData[len(db.data.CounterData)-1]
+}
+
+func (db profilingDataBuilder) addSample(sampler *CounterData, counter counterType, value float64) {
+	if _, ok := db.seenCounters[counter]; !ok {
+		db.seenCounters[counter] = struct{}{}
+		db.data.CounterSpecs = append(db.data.CounterSpecs, counters[counter])
+	}
+	sampler.Samples = append(sampler.Samples, Sample{uint32(counter), value})
+}
+
+// Construct a decoder for reading buffer memory
+func getBufferDecoder(ctx context.Context, s *api.GlobalState, buffer BufferObjectʳ, offset VkDeviceSize, size VkDeviceSize) (*memory.Decoder, error) {
+	// TODO: Not sure what to put for thread, but it doesn't seem to be used
```

Comment:
Do you need to resolve this TODO before submitting? Or turn it into a statement of why thread isn't used? Presumably it's one of the nils below?